pkg/gui/context.go

```diff
@@ -131,6 +131,35 @@ func (gui *Gui) popContext() error {
 	return gui.activateContext(newContext, types.OnFocusOpts{})
 }
 
+func (gui *Gui) removeAllPopupsFromContext() error {
+	gui.State.ContextManager.Lock()
+
+	if len(gui.State.ContextManager.ContextStack) == 1 {
+		gui.State.ContextManager.Unlock()
+		return nil
+	}
+
+	newContext := []types.Context{}
+	contextToDeactivate := []types.Context{}
+	for _, context := range gui.State.ContextManager.ContextStack {
```

Comment:
we can make this more concise by using slices.Partition (from my slices package)

```go
popups, rest := slices.Partition(gui.State.ContextManager.ContextStack, func(context *types.Context) bool {
	return context.GetKind() == types.TEMPORARY_POPUP || context.GetKind() == types.PERSISTENT_POPUP
})
gui.State.ContextManager.ContextStack = rest
contextToActivate := rest[len(rest)-1]
gui.State.ContextManager.Unlock()
```
It's also important that we write to `gui.State.ContextManager.ContextStack` before unlocking, to prevent race conditions